src/Features/Core/Portable/ConvertIfToSwitch/AbstractConvertIfToSwitchCodeRefactoringProvider.Analyzer.cs

```diff
@@ -0,0 +1,502 @@
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.ConvertIfToSwitch
+{
+    using static BinaryOperatorKind;
+
+    internal abstract partial class AbstractConvertIfToSwitchCodeRefactoringProvider
+    {
+        // Match the following pattern which can be safely converted to switch statement
+        //
+        //    <if-statement-sequence>
+        //        : if (<section-expr>) { <unreachable-end-point> }, <if-statement-sequence>
+        //        : if (<section-expr>) { <unreachable-end-point> }, ( return | throw )
+        //        | <if-statement>
+
+        //    <if-statement>
+        //        : if (<section-expr>) { _ } else <if-statement>
+        //        | if (<section-expr>) { _ } else { _ }
+        //        | if (<section-expr>) { _ }
+        //        | { <if-statement-sequence> }
+
+        //    <pattern-expr>
+        //        : <pattern-expr> && <expr>                         // C#
+        //        | <expr0> is <pattern>                             // C#
+        //        | <expr0> is <type>                                // C#
+        //        | <expr0> == <const-expr>                          // C#, VB
+        //        | <expr0> <comparison-op> <const>                  //     VB
+        //        | ( <expr0> >= <const> | <const> <= <expr0> )
+        //           && ( <expr0> <= <const> | <const> >= <expr0> )  //     VB
+        //        | ( <expr0> <= <const> | <const> >= <expr0> )
+        //           && ( <expr0> >= <const> | <const> <= <expr0> )  //     VB
+        //
+        //    <section-expr>
+        //        : <section-expr> || <pattern-expr>
+        //        | <pattern-expr>
+        //
+        internal abstract class Analyzer<TIfStatementSyntax> : IAnalyzer where TIfStatementSyntax : SyntaxNode
+        {
+            private SyntaxNode? _switchTargetExpression;
+            private readonly ISyntaxFactsService _syntaxFacts;
+
+            protected Analyzer(ISyntaxFactsService syntaxFacts)
+            {
+                _syntaxFacts = syntaxFacts;
+            }
+
+            public (ImmutableArray<SwitchSection>, SyntaxNode) AnalyzeIfStatementSequence(ReadOnlySpan<IOperation> operations)
+            {
+                var sections = ArrayBuilder<SwitchSection>.GetInstance();
+                if (!ParseIfStatementSequence(operations, sections, out var defaultBodyOpt))
+                {
+                    sections.Free();
+                    return default;
+                }
+
+                if (defaultBodyOpt is object)
+                {
+                    sections.Add(new SwitchSection(labels: default, defaultBodyOpt, defaultBodyOpt.Syntax));
+                }
+
+                Debug.Assert(_switchTargetExpression is object);
```

Comment:
Build is failing because _switchTargetExpression is nullable, but here we asserted that it's not.. isn't that sufficient?